packages/mongo/observe_handle.ts

```diff
@@ -15,11 +15,14 @@ export class ObserveHandle {
   nonMutatingCallbacks: boolean;
   _stopped: boolean;
 
-  _added?: (...args: any[]) => void;
-  _addedBefore?: (...args: any[]) => void;
-  _changed?: (...args: any[]) => void;
-  _movedBefore?: (...args: any[]) => void;
-  _removed?: (...args: any[]) => void;
+  public initialAddsSentResolver: (value: void) => void = () => {};
+  public initialAddsSent: Promise<void>
+
+  _added?: (...args: any[]) => Promise<void>;
+  _addedBefore?: (...args: any[]) => Promise<void>;
+  _changed?: (...args: any[]) => Promise<void>;
+  _movedBefore?: (...args: any[]) => Promise<void>;
+  _removed?: (...args: any[]) => Promise<void>;
```

Comment:
```suggestion
  _added?: (...args: any[]) => Promise<void> | void;
  _addedBefore?: (...args: any[]) => Promise<void> | void;
  _changed?: (...args: any[]) => Promise<void> | void;
  _movedBefore?: (...args: any[]) => Promise<void> | void;
  _removed?: (...args: any[]) => Promise<void> | void;
```
We could even do something like:

`class ObserveHandle<T = any>` and those methods be something like `(...args: T[]) => Promise<void> | void;`

And in the instance of the handle we could pass in the type of the collection(at least the name idk)